modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/util/OpenAPIParserUtil.java

```diff
@@ -449,25 +461,68 @@ public static boolean hasHTTPMethod(
 	}
 
 	private static void _getExternalReferences(
-		Map<String, Content> contents, Set<String> externalReferences) {
+		Map<String, Content> contents, Set<String> externalReferences,
+		Map<String, Schema> schemas) {
 
 		if (contents == null) {
 			return;
 		}
 
 		for (Content content : contents.values()) {
-			Schema schema = content.getSchema();
+			Schema contentSchema = content.getSchema();
 
-			if (schema == null) {
+			if (contentSchema == null) {
 				continue;
 			}
 
-			String reference = _getReference(schema);
-
-			if ((reference != null) &&
-				!reference.contains("#/components/schemas/")) {
-
-				externalReferences.add(reference);
+			Queue<Map<String, Schema>> queue = new LinkedList<>();
+
+			queue.add(Collections.singletonMap("content", contentSchema));
+
+			Map<String, Schema> map = null;
+			Set<String> visited = new HashSet<>();
+
+			while ((map = queue.poll()) != null) {
```

Comment:
Can we extract this somewhere? I think it's the third time with the same structure.

Reply:
In this case, I think that it is not so similar to the other 2 (much similar between them) and I think would be difficult to generalize for the 3 cases. May be we can give it a look to the other 2 to refactor them